### DISCUSSION TITLE
[QUESTION] Importing new loading information into an existing .ideaCon file.

### DISCUSSION THREAD

**Discussion by oatless** [idea-statica/ideastatica-public#198] [Help Q&A] [answered]
Hi,

I want to import new loading information into an existing connection and change the element names.
I am using ConnHiddenClientFactory to access the connection and I see that I can use GetConnectionLoadingJSON and UpdateLoadingFromJson to get and set the loadings for the connection. But I still have several questions:

1. What is the forcesPercentageOnSegments part in the LoadingJSON? Is it required?
2. What is the best way to generate the JSON file given a set of loads?
3. How can I change the names of elements in the connection?

Thanks in advance!

**Comment by nathanlukeIDEA** [comment]
Hi oatless,

Regarding point 1:
As you are probably aware, the program's forces can be defined as a percentage of the section capacity or by explicit force. 
The toggle within the JSON loading definition specifies which values are taken from the JSON input:- Either forcesOnSegments or forcesPercentageOnSegments. 
![image](https://user-images.githubusercontent.com/93578548/194792170-32f5b1c9-7d41-479b-bff7-0863acd15a81.png)
If the percentageload=false (ie. forces are not defined as percentages) you can leave these values as zero on the input.

Regarding Point 2:
Are you using c# or python? I may be able to provide more advice depending on how you are trying to achieve this. 

Regarding Point 3: 
You cannot currently change the names of Members in the connection using the API. If you don't mind can you elaborate on why you are wanting to do this?

Hope this helps.

Nathan

**Reply by oatless** [reply]
Hi Nathan,

Thanks for explaining the forcesPercentageOnSegments. I am using c# for this project. 
Regarding Point 3 - I want to match the member names in Idea to element numbers in the FEA Application. The plan is to import the geometry of the connection from Advance Steel and then import the loadings into the .ideaCon file from the FEA Application using the Idea API. To facilitate verification it would be useful if the members in Idea had the same names as in the FEA Application.

Thanks for your help!

**Comment by nathanlukeIDEA** [comment]
Hi oatless, 

Okay, thanks. If you really want to do this, it is possible to update an existing connection with a new IOM where the members are renamed. 

client.UpdateConProjFromIOM(.....)

You could then also input the loads through the IOM Results definition, which may be slightly easier. 

I am wondering if you would be willing to have a video discussion on this to see if we can help further?

Feel free to send me an email at nathan.luke@ideastatica.com

Cheers
Nathan

**Reply by oatless** [reply]
Hi Nathan,

Updating with a new IOM sounds interesting. Is there a way to export the current geometry in the file and read it back in as a new IOM? Would you be able to give me a quick summary of the process?

Thanks for your help!

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi oatless, 

Below would be the process based on the information you have provided.

```c#
public static class IdeaConUpdateHelper
{
    public static void updateExistingConnection(string ideaConPath, ConnectionHiddenCheckClient client)
    {
        //Open IdeaConnection
        client.OpenProject(ideaConPath);

        //Get first connection Id
        string id = client.GetProjectInfo().Connections[0].Identifier;

        //Get Idea IOM Model and IOMResult as OpenModelContainer
        string conContainerDataXML = client.GetAllConnectionData(id);

        //Deserialize the OpenModelContainer and Get the OpenModel component - Results are ignored.
        OpenModel model = IdeaStatiCa.Plugin.Tools.OpenModelContainerFromXml(conContainerDataXML).OpenModel;

        //Update Member Ids
        foreach (IdeaRS.OpenModel.Model.Member1D member in model.Member1D)
        {
            //find and update names based on your name map.
        }

        //Save the updated OpenModel to a Location
        string filepathToUpdatedIOM = "C:\\Desktop\\updated.xml";
        model.SaveToXmlFile(filepathToUpdatedIOM);

        //Build the Open Model Result From Your FEA application.
        OpenModelResult result = new OpenModelResult();

        //Result path
        string resultPath = "C:\\Desktop\\results.xml";
        result.SaveToXmlFile(resultPath);

        //update the project from new IOM and new results.
        client.UpdateConProjFromIOM(filepathToUpdatedIOM, resultPath);

        //Save As the project.
        client.SaveAsProject("newFilePath.ideacon");
    }
}
```

See if you think this is feasible... 

If you would prefer to update the loading directly on the connection file - as you first mentioned, I think we should make some improvements to allow for the addition of Load Effects before modifying load values through c#.

For your thought, It would likely be easier to do the member renaming (and IOM result creation) at the project level, i.e renaming all members in the IOM model before creating the connections, but I am not sure how you are creating your connection files or where your FEA results are coming from. If you want to discuss the possibilities let me know. 

Cheers
Nathan
